packages/fx-core/src/component/generator/apiPluginFromScratch/generator.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @author yuqzho@microsoft.com
+ */
+
+import { Context, FxError, Inputs, ok, Result } from "@microsoft/teamsfx-api";
+import { DefaultTemplateGenerator } from "../templates/templateGenerator";
+import {
+  ApiAuthOptions,
+  ApiPluginStartOptions,
+  CapabilityOptions,
+  ProgrammingLanguage,
+  QuestionNames,
+} from "../../../question";
+import { ActionContext } from "../../middleware/actionExecutionMW";
+import { Generator } from "../generator";
+import { merge } from "lodash";
+import { TemplateNames } from "../templates/templateNames";
+import { TemplateInfo } from "../templates/templateInfo";
+
+const enum telemetryProperties {
+  templateName = "template-name",
+  isDeclarativeCopilot = "is-declarative-copilot",
+}
+
+export class ApiPluginFromScratchGenerator extends DefaultTemplateGenerator {
```

Comment:
You may want to add a property named `componentName`. So that we can distinguish the telemetry events that are from this generator.